x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Dissect.java

```diff
@@ -52,6 +77,28 @@ protected NodeInfo<? extends LogicalPlan> info() {
         return NodeInfo.create(this, Dissect::new, child(), input, parser, extractedFields);
     }
 
+    @Override
+    public Dissect withGeneratedNames(List<String> newNames) {
+        if (newNames.size() != extractedFields.size()) {
+            throw new IllegalArgumentException(
+                "Number of new names is [" + newNames.size() + "] but there are [" + extractedFields.size() + "] existing names."
+            );
+        }
```

Comment:
Default method is simple enough! I'm afraid of changing the class hierarchy of `RegexExtract`, `Enrich` and `Eval` to put an abstract class in between there: this might, _maybe_, mess up some `instanceOf` checks, and we might have to fiddle with `EsqlNodeSubclassTests`, which I like to avoid.